--- ./occt/src/BRepAlgoAPI/BRepAlgoAPI_Algo.hxx	2020-10-04 10:27:14.616788017 +0200
+++ ./occt/src/BRepAlgoAPI/BRepAlgoAPI_Algo.hxx	2020-10-04 10:27:08.703429562 +0200
@@ -41,21 +41,36 @@
 
   // Provide access to methods of protected base class BOPAlgo_Options
   // (inherited as protected to avoid problems with SWIG wrapper)
-  using BOPAlgo_Options::Clear;
-  using BOPAlgo_Options::SetRunParallel;
-  using BOPAlgo_Options::RunParallel;
-  using BOPAlgo_Options::SetFuzzyValue;
-  using BOPAlgo_Options::FuzzyValue;
-  using BOPAlgo_Options::HasErrors;
-  using BOPAlgo_Options::HasWarnings;
-  using BOPAlgo_Options::HasError;
-  using BOPAlgo_Options::HasWarning;
-  using BOPAlgo_Options::DumpErrors;
-  using BOPAlgo_Options::DumpWarnings;
-  using BOPAlgo_Options::ClearWarnings;
-  using BOPAlgo_Options::GetReport;
-  using BOPAlgo_Options::SetProgressIndicator;
-  using BOPAlgo_Options::SetUseOBB;
+  // using BOPAlgo_Options::Clear;
+  void Clear() { return BOPAlgo_Options::Clear(); }
+  // using BOPAlgo_Options::SetRunParallel;
+  void SetRunParallel(const Standard_Boolean theFlag) { return BOPAlgo_Options::SetRunParallel(theFlag); }
+  // using BOPAlgo_Options::RunParallel;
+  Standard_Boolean RunParallel() const { return BOPAlgo_Options::RunParallel(); }
+  // using BOPAlgo_Options::SetFuzzyValue;
+  void SetFuzzyValue(const Standard_Real theFuzz) { return BOPAlgo_Options::SetFuzzyValue(theFuzz); }
+  // using BOPAlgo_Options::FuzzyValue;
+  Standard_Real FuzzyValue() const { return BOPAlgo_Options::FuzzyValue(); }
+  // using BOPAlgo_Options::HasErrors;
+  Standard_Boolean HasErrors() const { return BOPAlgo_Options::HasErrors(); }
+  // using BOPAlgo_Options::HasWarnings;
+  Standard_Boolean HasWarnings() const { return BOPAlgo_Options::HasWarnings(); }
+  // using BOPAlgo_Options::HasError;
+  Standard_Boolean HasError (const Handle(Standard_Type)& theType) const { return BOPAlgo_Options::HasError(theType); }
+  // using BOPAlgo_Options::HasWarning;
+  Standard_Boolean HasWarning (const Handle(Standard_Type)& theType) const { return BOPAlgo_Options::HasWarning(theType); }
+  // using BOPAlgo_Options::DumpErrors;
+  void DumpErrors(Standard_OStream& theOS) const { return BOPAlgo_Options::DumpErrors(theOS); }
+  // using BOPAlgo_Options::DumpWarnings;
+  void DumpWarnings(Standard_OStream& theOS) const { return BOPAlgo_Options::DumpWarnings(theOS); }
+  // using BOPAlgo_Options::ClearWarnings;
+  void ClearWarnings() { return BOPAlgo_Options::ClearWarnings(); }
+  // using BOPAlgo_Options::GetReport;
+  const Handle(Message_Report)& GetReport () const { return BOPAlgo_Options::GetReport(); }
+  // using BOPAlgo_Options::SetProgressIndicator;
+  void SetProgressIndicator(const Handle(Message_ProgressIndicator)& theObj) { return BOPAlgo_Options::SetProgressIndicator(theObj); }
+  // using BOPAlgo_Options::SetUseOBB;
+  void SetUseOBB(const Standard_Boolean theUseOBB) { return BOPAlgo_Options::SetUseOBB(theUseOBB); }
 
 protected:
 
