--- ./occt/src/BRepGProp/BRepGProp.hxx	2020-04-01 15:52:54.000000000 +0200
+++ ./occt/src/BRepGProp/BRepGProp.hxx	2020-04-24 22:05:49.239687279 +0200
@@ -160,6 +160,11 @@
   //! are taken into calculation only once.
   Standard_EXPORT static Standard_Real SurfaceProperties (const TopoDS_Shape& S, GProp_GProps& SProps,
                         const Standard_Real Eps, const Standard_Boolean SkipShared = Standard_False);
+  Standard_EXPORT static Standard_Real SurfaceProperties2 (const TopoDS_Shape& S, GProp_GProps& SProps,
+                        const Standard_Real Eps, const Standard_Boolean SkipShared = Standard_False) {
+    return SurfaceProperties(S, SProps, Eps, SkipShared);
+  }
+
   //!
   //! Computes the global volume properties of the solid
   //! S, and brings them together with the global
@@ -224,6 +229,11 @@
   Standard_EXPORT static Standard_Real VolumeProperties (const TopoDS_Shape& S, GProp_GProps& VProps, 
                          const Standard_Real Eps, const Standard_Boolean OnlyClosed = Standard_False, 
                                                  const Standard_Boolean SkipShared = Standard_False);
+  Standard_EXPORT static Standard_Real VolumeProperties2 (const TopoDS_Shape& S, GProp_GProps& VProps, 
+                         const Standard_Real Eps, const Standard_Boolean OnlyClosed = Standard_False, 
+                                                 const Standard_Boolean SkipShared = Standard_False) {
+    return VolumeProperties(S, VProps, Eps, OnlyClosed, SkipShared);
+  }
   
   //! Updates <VProps> with the shape <S>, that contains its pricipal properties.
   //! The volume properties of all the FORWARD and REVERSED faces in <S> are computed.
@@ -257,6 +267,16 @@
     const Standard_Boolean CGFlag = Standard_False, 
     const Standard_Boolean IFlag = Standard_False, 
     const Standard_Boolean SkipShared = Standard_False);
+  Standard_EXPORT static Standard_Real VolumePropertiesGK2 (const TopoDS_Shape& S, 
+    GProp_GProps& VProps, 
+    const gp_Pln& thePln, const Standard_Real Eps = 0.001, 
+    const Standard_Boolean OnlyClosed = Standard_False, 
+    const Standard_Boolean IsUseSpan = Standard_False, 
+    const Standard_Boolean CGFlag = Standard_False, 
+    const Standard_Boolean IFlag = Standard_False, 
+    const Standard_Boolean SkipShared = Standard_False) {
+      return VolumePropertiesGK(S, VProps, thePln, Eps, OnlyClosed, IsUseSpan, CGFlag, IFlag, SkipShared);
+    }
 
 
 protected:
