--- ./occt/src/OSD/OSD_Process.cxx	2019-09-29 11:12:58.000000000 +0200
+++ ./occt/src/OSD/OSD_Process.cxx	2020-04-16 23:52:58.284299351 +0200
@@ -99,8 +99,7 @@
 
 TCollection_AsciiString OSD_Process::UserName()
 {
-  struct passwd *anInfos = getpwuid (getuid());
-  return TCollection_AsciiString (anInfos ? anInfos->pw_name : "");
+  return TCollection_AsciiString ("");
 }
 
 Standard_Boolean OSD_Process::IsSuperUser (){
