diff -ruN /occt/original/occt-0dc2c37/src/AIS/AIS_Shape.hxx /occt/occt-0dc2c37/src/AIS/AIS_Shape.hxx
--- /occt/original/occt-0dc2c37/src/AIS/AIS_Shape.hxx	2021-02-02 08:51:56.000000000 +0000
+++ /occt/occt-0dc2c37/src/AIS/AIS_Shape.hxx	2021-02-14 13:25:00.243949219 +0000
@@ -170,7 +170,8 @@
   
   //! AIS_InteractiveObject defines another virtual method BoundingBox,
   //! which is not the same as above; keep it visible.
-  using AIS_InteractiveObject::BoundingBox;
+  // using AIS_InteractiveObject::BoundingBox;
+  void BoundingBox (Bnd_Box& theBndBox) override { AIS_InteractiveObject::BoundingBox(theBndBox); }
 
   //! Returns the Color attributes of the shape accordingly to
   //! the current facing model;
diff -ruN /occt/original/occt-0dc2c37/src/BlendFunc/BlendFunc_ChamfInv.hxx /occt/occt-0dc2c37/src/BlendFunc/BlendFunc_ChamfInv.hxx
--- /occt/original/occt-0dc2c37/src/BlendFunc/BlendFunc_ChamfInv.hxx	2021-02-02 08:51:56.000000000 +0000
+++ /occt/occt-0dc2c37/src/BlendFunc/BlendFunc_ChamfInv.hxx	2021-02-14 13:25:16.464009587 +0000
@@ -59,7 +59,8 @@
   //! False otherwise.
   Standard_EXPORT Standard_Boolean Derivatives (const math_Vector& X, math_Matrix& D) Standard_OVERRIDE;
 
-  using Blend_FuncInv::Set;
+  // using Blend_FuncInv::Set;
+  void Set (const Standard_Boolean OnFirst, const Handle(Adaptor2d_HCurve2d)& COnSurf) override { BlendFunc_GenChamfInv::Set(OnFirst, COnSurf); }
   
   Standard_EXPORT virtual void Set (const Standard_Real Dist1,
                                     const Standard_Real Dist2,
diff -ruN /occt/original/occt-0dc2c37/src/BlendFunc/BlendFunc_ConstThroatInv.hxx /occt/occt-0dc2c37/src/BlendFunc/BlendFunc_ConstThroatInv.hxx
--- /occt/original/occt-0dc2c37/src/BlendFunc/BlendFunc_ConstThroatInv.hxx	2021-02-02 08:51:56.000000000 +0000
+++ /occt/occt-0dc2c37/src/BlendFunc/BlendFunc_ConstThroatInv.hxx	2021-02-14 13:25:30.307394655 +0000
@@ -55,7 +55,8 @@
   //! False otherwise.
   Standard_EXPORT Standard_Boolean Derivatives (const math_Vector& X, math_Matrix& D) Standard_OVERRIDE;
   
-  using Blend_FuncInv::Set;
+  // using Blend_FuncInv::Set;
+  void Set (const Standard_Boolean OnFirst, const Handle(Adaptor2d_HCurve2d)& COnSurf) override { BlendFunc_GenChamfInv::Set(OnFirst, COnSurf); }
   
   Standard_EXPORT virtual void Set (const Standard_Real theThroat,
                                     const Standard_Real,
diff -ruN /occt/original/occt-0dc2c37/src/BRepAlgoAPI/BRepAlgoAPI_Algo.hxx /occt/occt-0dc2c37/src/BRepAlgoAPI/BRepAlgoAPI_Algo.hxx
--- /occt/original/occt-0dc2c37/src/BRepAlgoAPI/BRepAlgoAPI_Algo.hxx	2021-02-02 08:51:56.000000000 +0000
+++ /occt/occt-0dc2c37/src/BRepAlgoAPI/BRepAlgoAPI_Algo.hxx	2021-02-14 13:26:00.374174339 +0000
@@ -40,21 +40,36 @@
 
   // Provide access to methods of protected base class BOPAlgo_Options
   // (inherited as protected to avoid problems with SWIG wrapper)
-  using BOPAlgo_Options::Clear;
-  using BOPAlgo_Options::SetRunParallel;
-  using BOPAlgo_Options::RunParallel;
-  using BOPAlgo_Options::SetFuzzyValue;
-  using BOPAlgo_Options::FuzzyValue;
-  using BOPAlgo_Options::HasErrors;
-  using BOPAlgo_Options::HasWarnings;
-  using BOPAlgo_Options::HasError;
-  using BOPAlgo_Options::HasWarning;
-  using BOPAlgo_Options::DumpErrors;
-  using BOPAlgo_Options::DumpWarnings;
-  using BOPAlgo_Options::ClearWarnings;
-  using BOPAlgo_Options::GetReport;
-  using BOPAlgo_Options::SetProgressIndicator;
-  using BOPAlgo_Options::SetUseOBB;
+  // using BOPAlgo_Options::Clear;
+  // using BOPAlgo_Options::SetRunParallel;
+  // using BOPAlgo_Options::RunParallel;
+  // using BOPAlgo_Options::SetFuzzyValue;
+  // using BOPAlgo_Options::FuzzyValue;
+  // using BOPAlgo_Options::HasErrors;
+  // using BOPAlgo_Options::HasWarnings;
+  // using BOPAlgo_Options::HasError;
+  // using BOPAlgo_Options::HasWarning;
+  // using BOPAlgo_Options::DumpErrors;
+  // using BOPAlgo_Options::DumpWarnings;
+  // using BOPAlgo_Options::ClearWarnings;
+  // using BOPAlgo_Options::GetReport;
+  // using BOPAlgo_Options::SetProgressIndicator;
+  // using BOPAlgo_Options::SetUseOBB;
+  void Clear() override { BOPAlgo_Options::Clear(); }
+  void SetRunParallel(const Standard_Boolean theFlag) { BOPAlgo_Options::SetRunParallel(theFlag); }
+  Standard_Boolean RunParallel() const { return BOPAlgo_Options::RunParallel(); }
+  void SetFuzzyValue(const Standard_Real theFuzz) { BOPAlgo_Options::SetFuzzyValue(theFuzz); }
+  Standard_Real FuzzyValue() const { return BOPAlgo_Options::FuzzyValue(); }
+  Standard_Boolean HasErrors() const { return BOPAlgo_Options::HasErrors(); }
+  Standard_Boolean HasWarnings() const { return BOPAlgo_Options::HasWarnings(); }
+  Standard_Boolean HasError (const Handle(Standard_Type)& theType) const { return BOPAlgo_Options::HasError(theType); }
+  Standard_Boolean HasWarning (const Handle(Standard_Type)& theType) const { return BOPAlgo_Options::HasWarning(theType); }
+  void DumpErrors(Standard_OStream& theOS) const { BOPAlgo_Options::DumpErrors(theOS); }
+  void DumpWarnings(Standard_OStream& theOS) const { BOPAlgo_Options::DumpWarnings(theOS); }
+  void ClearWarnings() { BOPAlgo_Options::ClearWarnings(); }
+  const Handle(Message_Report)& GetReport () const { return BOPAlgo_Options::GetReport(); }
+  void SetProgressIndicator(const Message_ProgressScope& theProgress) { BOPAlgo_Options::SetProgressIndicator(theProgress); }
+  void SetUseOBB(const Standard_Boolean theUseOBB) { BOPAlgo_Options::SetUseOBB(theUseOBB); }
 
 protected:
 
diff -ruN /occt/original/occt-0dc2c37/src/Graphic3d/Graphic3d_Buffer.hxx /occt/occt-0dc2c37/src/Graphic3d/Graphic3d_Buffer.hxx
--- /occt/original/occt-0dc2c37/src/Graphic3d/Graphic3d_Buffer.hxx	2021-02-02 08:51:56.000000000 +0000
+++ /occt/occt-0dc2c37/src/Graphic3d/Graphic3d_Buffer.hxx	2021-02-14 13:26:26.004271348 +0000
@@ -179,8 +179,10 @@
 //! @name general accessors
 public:
 
-  using NCollection_Buffer::Data;
-  using NCollection_Buffer::ChangeData;
+  // using NCollection_Buffer::Data;
+  // using NCollection_Buffer::ChangeData;
+  const Standard_Byte* Data() const { return NCollection_Buffer::Data(); }
+  Standard_Byte* ChangeData() { return NCollection_Buffer::ChangeData(); }
 
   //! Return the attribute data with stride size specific to this attribute.
   //! @param theAttrib       attribute to find
diff -ruN /occt/original/occt-0dc2c37/src/V3d/V3d_DirectionalLight.hxx /occt/occt-0dc2c37/src/V3d/V3d_DirectionalLight.hxx
--- /occt/original/occt-0dc2c37/src/V3d/V3d_DirectionalLight.hxx	2021-02-02 08:51:56.000000000 +0000
+++ /occt/occt-0dc2c37/src/V3d/V3d_DirectionalLight.hxx	2021-02-14 13:26:40.194325311 +0000
@@ -38,7 +38,9 @@
 
   //! Defines the direction of the light source by a predefined orientation.
   Standard_EXPORT void SetDirection (V3d_TypeOfOrientation theDirection);
-  using Graphic3d_CLight::SetDirection;
+  // using Graphic3d_CLight::SetDirection;
+  void SetDirection (const gp_Dir& theDir) { Graphic3d_CLight::SetDirection(theDir); }
+  void SetDirection (Standard_Real theVx, Standard_Real theVy, Standard_Real theVz) { Graphic3d_CLight::SetDirection(theVx, theVy, theVz); }
 
 //! @name hidden properties not applicable to directional light
 private:
diff -ruN /occt/original/occt-0dc2c37/src/V3d/V3d_SpotLight.hxx /occt/occt-0dc2c37/src/V3d/V3d_SpotLight.hxx
--- /occt/original/occt-0dc2c37/src/V3d/V3d_SpotLight.hxx	2021-02-02 08:51:56.000000000 +0000
+++ /occt/occt-0dc2c37/src/V3d/V3d_SpotLight.hxx	2021-02-14 13:27:06.467759014 +0000
@@ -48,9 +48,15 @@
   //! Defines the direction of the light source
   //! according to a predefined directional vector.
   Standard_EXPORT void SetDirection (V3d_TypeOfOrientation theOrientation);
-  using Graphic3d_CLight::SetDirection;
-  using Graphic3d_CLight::Position;
-  using Graphic3d_CLight::SetPosition;
+  // using Graphic3d_CLight::SetDirection;
+  // using Graphic3d_CLight::Position;
+  // using Graphic3d_CLight::SetPosition;
+  void SetDirection (const gp_Dir& theDir) { Graphic3d_CLight::SetDirection(theDir); }
+  void SetDirection (Standard_Real theVx, Standard_Real theVy, Standard_Real theVz) { Graphic3d_CLight::SetDirection(theVx, theVy, theVz); }
+  const gp_Pnt& Position() const { return Graphic3d_CLight::Position(); }
+  void Position (Standard_Real& theX, Standard_Real& theY, Standard_Real& theZ) const { Graphic3d_CLight::Position(theX, theY, theZ); }
+  void SetPosition (const gp_Pnt& thePosition) { Graphic3d_CLight::SetPosition(thePosition); }
+  void SetPosition (Standard_Real theX, Standard_Real theY, Standard_Real theZ) { Graphic3d_CLight::SetPosition(theX, theY, theZ); }
 
 };
 
